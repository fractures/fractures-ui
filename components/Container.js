import React, { Component } from "react"

class Container extends Component {
	render() {
		return (
			<div className={ `fr-container mx-auto ${ this.props.className ? `${ this.props.className }` : `` }` }>
				{this.props.children}
				<style jsx global>{`
					@media (max-width: 1120px) {
						.fr-container {
							margin-left: 1.5rem !important;
							margin-right: 1.5rem !important;

							max-width: calc(100% - 3rem) !important;
						}
					}

					.fr-container {
						max-width: 100%;
						width: 1060px;
					}
				`}</style>
			</div>
		)
	}
}

export default Container
