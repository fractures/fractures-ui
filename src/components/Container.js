import PropTypes from "prop-types"
import React from "react"

const propTypes = {
	className: PropTypes.string,
	children: PropTypes.node.isRequired
}

const Container = props => (
	<div className={ `fr-container mx-auto ${ props.className ? `${ props.className }` : `` }` }>
		{props.children}
		<style jsx global>{`
			@media screen and (max-width: 1120px) {
				.fr-container {
					margin-left: 1.5rem !important;
					margin-right: 1.5rem !important;

					max-width: calc(100% - 3rem) !important;
				}
			}

			.fr-container {
				max-width: 100%;
				width: 1060px;
			}
		`}</style>
	</div>
)

Container.propTypes = propTypes

export default Container
