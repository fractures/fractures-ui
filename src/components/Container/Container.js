import PropTypes from "prop-types"
import styled from "styled-components"

const propTypes = {
	breakPoint: PropTypes.number,
	children: PropTypes.node.isRequired,
	className: PropTypes.string,
	width: PropTypes.number
}

const defaultProps = {
	breakPoint: 1060 + 48,
	width: 1060
}

const Container = styled.div`
	margin-left: auto;
	margin-right: auto;
	max-width: 100%;

	width: ${ props => props.width }px;

	@media screen and (max-width: ${ props => props.breakPoint }px) {
		margin-left: 1.5rem !important;
		margin-right: 1.5rem !important;
		max-width: calc(100% - 3rem) !important;
	}
`

Container.propTypes = propTypes
Container.defaultProps = defaultProps

export default Container
