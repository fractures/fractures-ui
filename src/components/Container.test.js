import Container from "./Container.js"

describe("Container", () => {
	const componentRender = (
		<Container>
			<div />
		</Container>
	)
	const component = renderer.create(componentRender)
	const tree = component.toJSON()

	it("Renders correctly", () => {
		expect(tree).toMatchSnapshot()
	})
})
