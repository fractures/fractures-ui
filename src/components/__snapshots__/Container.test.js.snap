// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Renders correctly 1`] = `
<div
  className="jsx-2434684429 fr-container mx-auto "
>
  <div />
</div>
`;
